test/pkg/test/framework.go

```diff
@@ -145,7 +145,7 @@ func (env *Framework) ExpectToBeClean(ctx context.Context) {
 		retrievedTargetGroups, _ := env.LatticeClient.ListTargetGroupsAsList(ctx, &vpclattice.ListTargetGroupsInput{})
 		for _, tg := range retrievedTargetGroups {
 			Logger(ctx).Infof("Found TargetGroup: %s, checking it whether it's created by current EKS Cluster", *tg.Id)
-			if currentClusterVpcId != *tg.VpcIdentifier {
+			if tg.VpcIdentifier != nil && currentClusterVpcId != *tg.VpcIdentifier {
```

Comment:
did you have issues with vpcIndetifier being nil?

Reply:
Yes, it caused e2e tests to fail because of a Lambda TG in my account (unrelated to EKS). The e2e tests should not let unrelated resources interfere with them, with the exception of service quotas.

Reply:
cool, as long it's not caused by refactoring